client/rest-high-level/src/test/java/org/elasticsearch/client/SecurityIT.java

```diff
@@ -222,7 +222,7 @@ private PutUserRequest randomPutUserRequest(boolean enabled) {
     }
 
     private static PutUserRequest randomPutUserRequest(User user, boolean enabled) {
-        final char[] password = randomAlphaOfLengthBetween(6, 10).toCharArray();
+        final char[] password = randomAlphaOfLengthBetween(14, 19).toCharArray();
```

Comment:
This is not only about verification. Put user requests will trigger password hashing, which will fail in fips mode with short passwords

Reply:
It doesn't look like it will. PutUserRequestBuilder does trigger a hasher, but we never get there. Looks like PutUserRequest is only used in tests.
Whith this said, I don't think it's a big problem to leave {14, 19} range here as long as we still have some tests covering the short allowed passwords when Fips is off.

Reply:
> It doesn't look like it will.

It will. 

Try `./gradlew client:rest-high-level:integTest --tests="org.elasticsearch.client.SecurityIT.testPutUser" in this branch which fails with passwords < 14 characters.

Reply:
Worked for me... Yes, I set the renge back to 6-10...

Reply:
This test suite sets up a test cluster (client/rest-high-level/build.gradle)  and `SecurityIT` performs security API related tests with the high level client against that cluster. `SecurityIT#testPutUser` creates a random PutUserRequest and _sends_ that request to the test cluster via the HLRC ( `SecurityClient securityClient = highLevelClient().security();` )
When the test cluster processes this request it will hash that user password in the process of storing it to the security index ( see `RestPutUserAction#innerPrepareRequest` which triggers the hashing ) . This hashing will fail when PBKDF2 is used and an exception will be thrown. 

If you don't get the failure, you either not run in FIPS mode ( `-Dtests.fips.enabled=true` ) or you don't run the test from this branch. You cannot try this in i.e. master as master doesn't yet contain [the fix](https://github.com/elastic/elasticsearch/pull/64024/files#diff-81f5504fc9e931a18720c9d0ecc27c0d9f610961e6e7c8e1fc072cfa0a11e5f0R56) which is the reason for this PR and sets the test cluster correctly in fips mode ) 

Hope this makes more sense now

Reply:
We agreed on why this change in the test is necessary over a chat, as such I'm resolving the comment